gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/TokenStateDatabase.java

```diff
@@ -41,6 +48,33 @@
 
   TokenStateDatabase(DataSource dataSource) throws Exception {
     this.dataSource = dataSource;
+    createKnoxTokensTableIfNotExists();
+  }
+
+  private void createKnoxTokensTableIfNotExists() throws Exception {
```

Comment:
Just curious what happens if the table creation throws an Exception.

Reply:
The Knox Gateway won't start. IMO, this is correct since table creation takes places only if token state service is explicitly set to JDBC and in this case, it's a must that the table is created in the configured DB.

Reply:
+1